packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceCreateManifest.test.ts

```diff
@@ -0,0 +1,175 @@
+const class1 = 'Apex1';
+const class2 = 'Apex2';
+const uri1 = vscode.Uri.parse(classPath + class1);
+const uri2 = vscode.Uri.parse(classPath + class2);
+const manifest = `<?xml version="1.0" encoding="UTF-8"?>
+<Package xmlns="http://soap.sforce.com/2006/04/metadata">
+  <types>
+    %s
+    <name>ApexClass</name>
+  </types>
+</Package>`;
+
+describe('Force Create Manifest', () => {
+  describe('Happy Paths', () => {
+
+    let env: sinon.SinonSandbox;
+    let openDocSpy: sinon.SinonSpy;
+    let inputBoxSpy: sinon.SinonStub;
+
+    before(() => {
+      env = createSandbox();
+    });
+
+    beforeEach(() => {
+      inputBoxSpy = env.stub(vscode.window, 'showInputBox');
+      openDocSpy = env.spy(vscode.workspace, 'openTextDocument');
+    });
+
+    afterEach(() => {
+      env.restore();
+    });
+
+    it('Should create first manifest from single sourceUri', async () => {
+      const packageXML = util.format(manifest, `<member>${class1}</member>`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync').returns(false);
+      env.stub(fs, 'mkdirSync').returns(undefined);
+      env.stub(fs, 'writeFileSync').returns(undefined);
+      const packageName = 'package' + generateRandomSuffix() + '.xml';
+      inputBoxSpy.onCall(0).returns(packageName);
+      await forceCreateManifest(uri1, undefined);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
+    });
+
+    it('Should create manifest from list of uris', async () => {
+      const packageXML = util.format(manifest, `<member>${class1}</member>\n<member>${class2}</member>\n`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync')
+        .onFirstCall()
+        .returns(true)
+        .onSecondCall()
+        .returns(false);
+      env.stub(fs, 'writeFileSync').returns(undefined);
+      const packageName = 'package' + generateRandomSuffix() + '.xml';
+      inputBoxSpy.onCall(0).returns(packageName);
+      await forceCreateManifest(uri1, [uri1, uri2]);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
+    });
+
+    it('Should create but not save manifest if cancelled', async () => {
+      const writeFileSpy = env.spy(fs, 'writeFileSync');
+      const packageXML = util.format(manifest, `<member>${class1}</member>\n<member>${class2}</member>\n`);
+      env.stub(ComponentSet, 'fromSource').returns({
+        getPackageXml: () => {
+          return packageXML;
+        }
+      });
+      env.stub(fs, 'existsSync')
+        .onFirstCall()
+        .returns(true)
+        .onSecondCall()
+        .returns(false);
+      inputBoxSpy.onCall(0).returns(undefined);
+      await forceCreateManifest(uri1, [uri1, uri2]);
+
+      expect(openDocSpy.calledOnce).to.equal(true);
+      expect(writeFileSpy.called).to.equal(false);
+    });
```

Comment:
Two test cases to consider: 
1. When the user doesn't supply a name use the default. 
2. When there are no entries in package.xml (like when you do it for a file in the manifests directory it produces an empty file.